### PR TITLE
[TEVA-859] Add system alerts to Prometheus and Alertmanager

### PR DESCRIPTION
**Pull request by nickromneydfe**
## Jira ticket URL

https://dfedigital.atlassian.net/browse/TEVA-859

## Changes in this PR:

- Renamed `files` as `config`
- Added file suffixes for clarity
- Supported sending either Slack URL and channel OR Slack config to `prometheus_all` module
- Used new location of `prometheus_all` terraform module
- Defined 5 alert rules, with associated Runbook and link to Grafana dashboard
- Used new Slack webhook for `#twd_tv_dev`

### CODE REVIEW

**Review by sfawcett123 — COMMENTED** [review]
Have you got a URL where this is running?

File: documentation/alert-runbook.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+### CloudFoundry
+
+Further information on setting up and logging in to CloudFoundry are in the [hosting](../hosting.md) document.
+
+- List apps with `cf apps`.
+- Check if any apps are crashed (e.g. `processes` may list 3 of 4 apps running: `web:3/4`)
+- Get the logs for the app
+```bash
+cf logs teaching-vacancies-production
+```
```
Comment (saliceti) [inline documentation/alert-runbook.md:37]: Can you suggest:
- restart app
- scale up number of instances, memory, disk...

File: documentation/alert-runbook.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+Alerts when either the `teaching-vacancies-production` or `teaching-vacancies-worker-production` app has an average Memory utilization above 60% for a 5-minute period.
+
+## ProdAppsCrashed
+
+Alerts when either the `teaching-vacancies-production` or `teaching-vacancies-worker-production` app reports a crash.
+
+## Action
+
+[Check the Grafana panel for an indication of what's going on.](https://grafana-teaching-vacancies.london.cloudapps.digital/d/6Ac4lUWGk/teaching-vacancies-production?orgId=1&refresh=5s)
```
Comment (saliceti) [inline documentation/alert-runbook.md:26]: Can we suggest:
- check rollbar
- check papertrail

File: terraform/monitoring/resources.tf [hunk 3]
```diff
@@ -1,13 +1,17 @@
...
-  alertmanager_config      = file("${path.module}/files/alertmanager.yml")
   grafana_admin_password   = local.secrets["grafana_admin_password"]
-  grafana_json_dashboards  = [file("${path.module}/files/paas_dashboard.json")]
-  alert_rules              = file("${path.module}/files/alert.rules")
+  grafana_json_dashboards  = [file("${path.module}/config/paas_dashboard.json")]
+  alert_rules              = file("${path.module}/config/alert.rules.yml")
+  # Send alertmanager_slack_channel and alertmanager_slack_url to prometheus_all to use default notifications
+  alertmanager_slack_url     = local.alertmanager_slack_url
+  alertmanager_slack_channel = local.alertmanager_slack_channel
+  # Send alertmanager_config to prometheus_all to use notifications defined in config/alertmanager.yml.tmpl
```
Comment (saliceti) [inline terraform/monitoring/resources.tf:15]: This can be removed

File: terraform/monitoring/variables.tf [hunk 4]
```diff
@@ -3,12 +3,18 @@ variable paas_user { default = null }
...
+  monitoring_instance_name   = local.service_name
+  paas_api_url               = "https://api.london.cloud.service.gov.uk"
+  monitoring_org_name        = "dfe-teacher-services"
+  space_name                 = "${local.service_name}-monitoring"
+  monitoring_space_name      = "${local.service_name}-monitoring"
+  aws_region                 = "eu-west-2"
+  secrets                    = yamldecode(data.aws_ssm_parameter.monitoring_secrets.value)
+  alertmanager_slack_url     = local.secrets["alertmanager_slack_url"]
+  alertmanager_slack_channel = "twd_tv_dev"
+  alertmanager_config = templatefile("${path.module}/config/alertmanager.yml.tmpl", {
```
Comment (saliceti) [inline terraform/monitoring/variables.tf:16]: This can be removed

File: terraform/monitoring/config/alertmanager.yml.tmpl [hunk 5]
```diff
@@ -0,0 +1,37 @@
+global:
```
Comment (saliceti) [inline terraform/monitoring/config/alertmanager.yml.tmpl:1]: This can be removed

File: terraform/monitoring/resources.tf (hunk 3 above)
Comment (nickromneydfe) [reply terraform/monitoring/resources.tf:15]: We still need to send `alertmanager_slack_url` and `alertmanager_slack_channel` to `prometheus_all` for the alertmanager module template to be populated

File: terraform/monitoring/variables.tf (hunk 4 above)
Comment (nickromneydfe) [reply terraform/monitoring/variables.tf:16]: I think lines 14 and 15 should still stand, as we need to send the teaching-vacancies specific values, but line 16 can be removed

File: terraform/monitoring/config/alertmanager.yml.tmpl (hunk 5 above)
Comment (nickromneydfe) [reply terraform/monitoring/config/alertmanager.yml.tmpl:1]: Agreed. I've `git rm`'d the whole file

**Review by saliceti — APPROVED** [review]

**Pull request merged**